### PR TITLE
Rewrite test generation to use new test format in definitions

### PR DESCRIPTION
**Pull request by ppeble**
Matching definitions PR: https://github.com/holidays/definitions/pull/26

I will need to point the definitions to master but that cannot happen until the linked PR is merged in the other repository.

### CODE REVIEW

**Comment by ppeble** [comment]
For potential reviewers of this PR: even though the generated code looks a lot different it is the same test content. What changed was the format of the tests in the definitions and how we read and generate the tests. There should be no behavior change.

**Comment by ppeble** [comment]
Whoops! I guess this won't build until I push to master. So I'll leave things 'red' until I merge in the linked PR (which should be soon). @ttwo32

File: .travis.yml [hunk 1]
```diff
@@ -18,7 +18,6 @@ notifications:
     holidaysgem@gmail.com
 
 rvm:
-  - 2.1.0
   - 2.2.0
```
Comment (ttwo32) [inline .travis.yml]: Thanks!!:)

File: lib/generated_definitions/north_america.rb [hunk 2]
```diff
@@ -158,7 +158,7 @@ def self.custom_methods
 
 "christmas_eve_holiday(date)" => Proc.new { |date|
 beginning_of_month = Date.civil(date.year, date.month, 1)
-(date.saturday? || date.sunday?) ? date.downto(beginning_of_month).find {|date| date if date.wday == 5} : date
+(date.saturday? || date.sunday?) ? date.downto(beginning_of_month).find {|d| d if d.wday == 5} : date
```
Comment (ttwo32) [inline lib/generated_definitions/north_america.rb]: Good!!
I like this change.:)
It's easier to read code than before.

File: lib/generated_definitions/us.rb [hunk 3]
```diff
@@ -92,7 +92,7 @@ def self.custom_methods
       {
         "christmas_eve_holiday(date)" => Proc.new { |date|
 beginning_of_month = Date.civil(date.year, date.month, 1)
-(date.saturday? || date.sunday?) ? date.downto(beginning_of_month).find {|date| date if date.wday == 5} : date
+(date.saturday? || date.sunday?) ? date.downto(beginning_of_month).find {|d| d if d.wday == 5} : date
```
Comment (ttwo32) [inline lib/generated_definitions/us.rb]: ❤️

File: test/holidays/definition/decorator/test_test.rb [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      :holiday? => true,
+    )
+
+    source = @decorator.call(entity)
+    expected_source = "assert_equal \"Test Holiday\", (Holidays.on(Date.civil(2016, 1, 1), [:us], [:option1])[0] || {})[:name]\n"
+
+    assert_equal expected_source, source
+  end
+
+  def test_call_generates_source_from_entity_single_date_with_holiday_flag
```
Comment (ttwo32) [inline test/holidays/definition/decorator/test_test.rb]: Nice test case!!:)

**Pull request merged**